tests/test_observer.py

```diff
@@ -0,0 +1,123 @@
+import enum
+import unittest
+import datetime
+
+from core_lib.core_lib import CoreLib
+from core_lib.observer.observer import Observer
+from core_lib.observer.observer_listener import ObserverListener
+from core_lib.observer.observer_registry import ObserverRegistry
+from core_lib.observer.observer_decorator import Observe
+
+
+class MyEnum(enum.Enum):
+    one = 1
+    two = 2
+    three = 3
+
+
+class UserObserverListener(ObserverListener):
+    EVENT = "EVENT"
+    last_key = ""
+    last_value = ""
+
+    def update(self, key: str, value):
+        UserObserverListener.last_key = key
+        UserObserverListener.last_value = value
+
+
+# with CoreLib Decorator
+class TestObserverWithDecorator(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        main_observer = Observer(UserObserverListener())
+        CoreLib.observer_registry.register("main", main_observer)
+
+    def test_observer_string(self):
+        self.set_observer("observe")
+        self.assertEqual("observe", UserObserverListener.last_value['var'])
+        self.set_observer("hello world")
+        self.assertEqual("hello world", UserObserverListener.last_value['var'])
+
+    def test_observer_float(self):
+        self.set_observer(14)
+        self.assertEqual(UserObserverListener.last_value['var'], 14)
+        self.set_observer(14.0987)
+        self.assertEqual(UserObserverListener.last_value['var'], 14.0987)
+
+    def test_observer_bool(self):
+        self.set_observer(True)
+        self.assertEqual(UserObserverListener.last_value['var'], True)
+        self.set_observer(False)
+        self.assertEqual(UserObserverListener.last_value['var'], False)
+
+    def test_observer_enum(self):
+        self.set_observer(MyEnum.one)
+        self.assertEqual(UserObserverListener.last_value['var'], MyEnum.one)
+        self.set_observer(MyEnum.two)
+        self.assertEqual(UserObserverListener.last_value['var'], MyEnum.two)
+        self.set_observer(MyEnum.three)
+        self.assertEqual(UserObserverListener.last_value['var'], MyEnum.three)
+
+    def test_observer_object(self):
+        dat = datetime.date(2022, 1, 1)
+        dct = {
+            'date': dat,
+            'tuple': ("fruit", "apple"),

```

Comment:
writing the value in the input and then again in the validation is error-prone. 
please create variables for values that repeat them selfs more the one time

suggestion 

`tuple_value = ("fruit", "apple")`